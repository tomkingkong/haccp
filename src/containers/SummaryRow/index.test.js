import React from 'react';
import { shallow } from 'enzyme';

import { SummaryRow } from '.';

describe('SummaryRow', () => {
  it('should match snapshot', () => {
    const mockIngredient = { name:'carrot' }
    const wrapper = shallow(<SummaryRow ingredient={mockIngredient}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with details', () => {
    const mockIngredient = {
      name: 'true',
      receiving_dna: true,
      receiving_letter: true,
      receiving_hazard_bio: true,
      receiving_hazard_chem: true,
      receiving_hazard_phys: true,
      receiving_hazard_handling: 'true',
      receiving_from: 'true',
      receiving_other: 'true',
      inventory_dna: true,
      inventory_hazard_bio: true,
      inventory_hazard_chem: true,
      inventory_hazard_phys: true,
      inventory_hazard_handling: 'true',
      inventory_other: 'true',
      processing_dna: true,
      processing_hazard_bio: true,
      processing_hazard_chem: true,
      processing_hazard_phys: true,
      processing_hazard_handling: 'true',
      processing_other: 'true',
      packaging_dna: true,
      packaging_hazard_bio: true,
      packaging_hazard_chem: true,
      packaging_hazard_phys: true,
      packaging_hazard_handling: 'true',
      packaging_method: 'true',
      packaging_other: 'true'
    }
    const wrapper = shallow(<SummaryRow ingredient={mockIngredient}/>);
    expect(wrapper).toMatchSnapshot();
  });
});


