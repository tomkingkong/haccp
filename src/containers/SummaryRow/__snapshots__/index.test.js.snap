// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryRow should match snapshot 1`] = `
<tr>
  <td>
    carrot
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      />
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Letter of Auth on file:
      </p>
      <p
        className="summary-info-single-line"
      >
        No
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Received From:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      />
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Received From:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      />
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      />
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        no
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Packaging method:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      />
    </div>
  </td>
</tr>
`;

exports[`SummaryRow should match snapshot with details 1`] = `
<tr>
  <td>
    true
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Does not apply
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Letter of Auth on file:
      </p>
      <p
        className="summary-info-single-line"
      >
        Yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Received From:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Does not apply
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Received From:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Does not apply
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
  </td>
  <td>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Does not apply
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Bio Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Chem Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-bool"
    >
      <p
        className="summary-info-single-line"
      >
        Physical Hazard:
      </p>
      <p
        className="summary-info-single-line"
      >
        yes
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Hazard Handling:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Packaging method:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
    <div
      className="summary-info-dual-line"
    >
      <p
        className="summary-table-item summary-table-item-header"
      >
        Other:
      </p>
      <p
        className="summary-table-item"
      >
        true
      </p>
    </div>
  </td>
</tr>
`;
